honeybee_radiance/dynamic/group.py

```diff
@@ -289,6 +291,46 @@ def states_json_list(self):
                 })
         return states_list
 
+    def rfluxmtx_control_params(self, sampling=None, up_direction=None):
+        """Get Rfluxmtx control parameters.
+
+        If the values are not provided they will be assigned based on the first State
+        of the dynamic object.
+
+        Args:
+            sampling: Set hemisphere sampling type. Acceptable inputs for hemisphere
+                sampling type are:
+                * u for uniform.(Usually applicable for ground).
+                * kf for klems full.
+                * kh for klems half.
+                * kq for klems quarter.
+                * rN for Reinhart - Tregenza type skies. N stands for subdivisions and
+                defaults to 1.
+                * scN for shirley-chiu subdivisions.
+                Add a ``-`` in front of the input for left-handed coordinates. For more
+                information see rfluxmtx docs.
+                https://www.radiance-online.org/learning/documentation/manual-pages/pdfs/rfluxmtx.pdf/at_download/file
+            up_direction: Orient the "up" direction for the hemisphere using the indicated
+                axis or direction vector. Valid inputs are [-]{X|Y|Z|ux,uy,uz}. Default: Y
+
+        Returns:
+            RfluxmtxControlParameters
+
+        """
+        state = self.states_by_index(0)[0]
+        if up_direction is None:
+            up_direction = state.vmtx_geometry.plane.y
+        if sampling is None:
+            try:
+                sampling = state.modifier.sampling_type
+            except AttributeError:
+                raise ValueError(
+                    'Rfluxmtx control parameters can only be generated for states '
+                    'with BSDF modifier. Current modifier: %s' % state.modifier
+                )
```

Comment:
Does this mean that we only support window groups with BSDF modifiers? If so, then this probably merits a check when we assign a `group_identifier` to an aperture to make sure that the Aperture has a BSDF assigned to it. It's better for users to get the error when they are creating the window group rather than getting all of the way to running the recipe and having it fail with a denser luigi/argo message.

Also, you should probably do `type(state.modifier)` or `state.modifier.__class__.__name__` since the repr of Modifiers isn't going to be that clear in an error message like this. It's probably also helpful for you to include the identifier of the modifier. Like:

```
Apertures in Window Groups must have a BSDF modifier. Current modifier "{}" is a {}.'.format(self.modifier, self.modifier.__class__.__name__)
```

Reply:
Thank you @chriswmackey! I made the adjustment to the message to be more clear.

> Does this mean that we only support window groups with BSDF modifiers?

The short answer is no. We support aperture groups with non-BSDF materials but in a 2-Phase recipe and not a 3 or 5 Phase which is when this error will be raised.

I agree that we should give an error sooner than later. Since this error is recipe-specific, the best place to raise an error is probably the model handler of the recipes. We can run a quick check on the model and raise an error before creating the run.